include/libdnf/transaction/transaction.hpp

```diff
@@ -0,0 +1,221 @@
+    void set_rpmdb_version_end(const std::string & value) { rpmdb_version_end = value; }
+
+    /// Get $releasever variable value that was used during the transaction
+    ///
+    /// @replaces libdnf:transaction/Transaction.hpp:method:Transaction.get_releasever()
+    const std::string & get_releasever() const noexcept { return releasever; }
+
+    /// Set $releasever variable value that was used during the transaction
+    ///
+    /// @replaces libdnf:transaction/private/Transaction.hpp:method:Transaction.setReleasever(const std::string & value)
+    void set_releasever(const std::string & value) { releasever = value; }
+
+    /// Get UID of a user that started the transaction
+    ///
+    /// @replaces libdnf:transaction/Transaction.hpp:method:Transaction.get_user_id()
+    uint32_t get_user_id() const noexcept { return user_id; }
+
+    /// Set UID of a user that started the transaction
+    ///
+    /// @replaces libdnf:transaction/private/Transaction.hpp:method:Transaction.setUserId(uint32_t value)
+    void set_user_id(uint32_t value) { user_id = value; }
+
+    /// Get command line of a program that started the transaction
+    ///
+    /// @replaces libdnf:transaction/Transaction.hpp:method:Transaction.get_cmdline()
+    const std::string & get_cmdline() const noexcept { return cmdline; }
+
+    /// Set command line of a program that started the transaction
+    ///
+    /// @replaces libdnf:transaction/private/Transaction.hpp:method:Transaction.setCmdline(const std::string & value)
+    void set_cmdline(const std::string & value) { cmdline = value; }
+
+    /// Get a user-specified comment describing the transaction
+    const std::string & get_comment() const noexcept { return comment; }
+
+    /// Set a user-specified comment describing the transaction
+    void set_comment(const std::string & value) { comment = value; }
+
+    /// Get transaction state
+    ///
+    /// @replaces libdnf:transaction/Transaction.hpp:method:Transaction.getState()
+    State get_state() const noexcept { return state; }
+
+    /// Set transaction state
+    ///
+    /// @replaces libdnf:transaction/private/Transaction.hpp:method:Transaction.setState(libdnf::TransactionState value)
+    void set_state(State value) { state = value; }
+
+    const std::set<std::string> & get_runtime_packages() const { return runtime_packages; }
+    void add_runtime_package(const std::string & nevra) { runtime_packages.insert(nevra); }
+
+    /// Get lines recorded during the transaction
+    const std::vector<std::pair<int, std::string>> & get_console_output() const { return console_output; }
+
+    /// Record a line printed during the transction to the database and also store it in the object
+    ///
+    /// The method must be called only on a started transaction
+    void add_console_output_line(int file_descriptor, const std::string & line);
+
+    void begin();
```

Comment:
I'm thinking of different name, because `begin()` is usually used for iterators. And I don't want to close doors for implementing standard iterators. May it be `start()`? Or something else?